app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -170,49 +171,49 @@ class ScreenController {
         // Call show() before hide() so that focus moves correctly to the shown fragment once others are hidden
         when (transition) {
             Transition.ADD_OVERLAY -> {
-                fragmentManagerShowNavigationOverlay(fragmentManager, true)
                 _currentActiveScreen.value = ActiveScreen.NAVIGATION_OVERLAY
```

Comment:
Add a comment (maybe above `when`, next to first currentActiveScreen assignment, or wrap currentActiveScreen assignment and comment in that method) that we need to update the active screen before we perform the fragment transactions so this behavior that we rely on doesn't regress.

Can we add a test for this?

Reply:
This does seem like a good thing to test, but it's not clear to me at the moment how.  Do you have any suggestions?

Reply:
First thought: if somehow you could stub `setCurrentActiveScreen` and `fragmentManagerShowNavigationOverlay`, you can have each toggle a boolean flag and check that the other boolean flag is in the appropriate state. It seems obnoxious to decouple these invocations so heavily, however.